textsplitter/split_documents.go

```diff
@@ -39,27 +43,36 @@ func CreateDocuments(textSplitter TextSplitter, texts []string, metadatas []map[
 
 	documents := make([]schema.Document, 0)
 
+	l := sync.Mutex{}
+	g, _ := errgroup.WithContext(context.Background())
+	g.SetLimit(int(math.Ceil(float64(len(texts)) / 10.0)))
```

Comment:
@iwilltry42 I re-think about it and think it might be better to set a default thread(number of goroutine limit) so user knows how many concurrent thread it will be running to split documents.